src/plugins/kibana_utils/common/url/encode_uri_query.ts

```diff
@@ -43,15 +43,38 @@ export function encodeUriQuery(val: string, pctEncodeSpaces = false) {
 
 export const encodeQuery = (
   query: ParsedQuery,
-  encodeFunction: (val: string, pctEncodeSpaces?: boolean) => string = encodeUriQuery
-) =>
-  transform(query, (result: any, value, key) => {
+  encodeFunction: (val: string, pctEncodeSpaces?: boolean) => string = encodeUriQuery,
+  pctEncodeSpaces = true
+): ParsedQuery =>
+  transform<any, ParsedQuery>(query, (result, value, key) => {
     if (key) {
       const singleValue = Array.isArray(value) ? value.join(',') : value;
 
       result[key] = encodeFunction(
         singleValue === undefined || singleValue === null ? '' : singleValue,
-        true
+        pctEncodeSpaces
       );
     }
   });
+
+/**
+ * Method to help modify url query params.
+ *
+ * @param params
+ * @param key
+ * @param value
+ */
+export const modifyParams = (params: string, key: string, value?: string) => {
+  const queryParams = parse(params);
+
+  if (value) {
+    queryParams[key] = value;
+  } else {
+    delete queryParams[key];
+  }
+
+  return stringify(encodeQuery(queryParams, undefined, false), {
+    sort: false,
+    encode: false,
```

Comment:
Why are we not encoding here actually?

Reply:
it's ok. For `kibana` urls we encode not all characters to make our url's a little bit more user friendly. In  `src/plugins/kibana_utils/common/url/encode_uri_query.ts` file you can find more detailed description.
So, to avoid of double encoding `encode:false` was added